### PR TITLE
refactor: Deprecated in 1.1: Occ configurable endpoints (GH-4125)

### LINKED ISSUE #4125: Deprecated in 1.1: Occ configurable endpoints (#4066)
For 2.0:

`client-authentication-token.service.ts`
- Make OccEndpointsService in constructor mandatory

`user-authentication-token.service.ts`
- Make OccEndpointsService in constructor mandatory

`occ-cart-entry.adapter.ts`
- Remove `getCartEndpoint`
- Remove //Todo 2.0: Remove
- Remove legacy methods
- Remove FeatureConfigService from constructor

`occ-cart.adapter.ts`
- Remove `getCartEndpoint`
- Remove //Todo 2.0: Remove
- Remove legacy methods
- Remove FeatureConfigService from constructor

`occ-user-order.adapter.ts`
- Remove `getCartEndpoint`
- Remove //Todo 2.0: Remove
- Remove legacy methods
- Remove FeatureConfigService from constructor

`feature-toggles.ts`
- Remove feature toggle for configurableOccEndpoints

### PR DESCRIPTION
**Pull request by bgambocjaviniar**
closes GH-4125

### CODE REVIEW

File: projects/schematics/src/migrations/2_0/constructor-deprecations/constructor-deprecation-data.ts [hunk 1]
```diff
@@ -18,11 +18,16 @@ import { CHECKOUT_PAYMENT_SERVICE_MIGRATION } from './data/checkout-payment.serv
 import { CHECKOUT_PROGRESS_MOBILE_TOP_COMPONENT_MIGRATION } from './data/checkout-progress-mobile-top.migration';
 import { CHECKOUT_GUARD_MIGRATIONS } from './data/checkout.guard.migration';
 import { CHECKOUT_SERVICE_MIGRATION } from './data/checkout.service.migration';
+import { CLIENT_AUTHENTICATION_TOKEN_SERVICE_MIGRATION } from './data/client-authentication-token.service.migration';
 import { CONFIGURABLE_ROUTES_SERVICE_MIGRATION } from './data/configurable-routes.service.migration';
 import { CURRENT_PRODUCT_SERVICE_MIGRATION } from './data/current-product-service.migration';
 import { EXTERNAL_ROUTES_SERVICE_MIGRATION } from './data/external-routes.service.migration';
+import { LOGIN_FORM_COMPONENT_MIGRATION } from './data/login-form.component.migration';
```
Comment (tobi-or-not-tobi) [inline projects/schematics/src/migrations/2_0/constructor-deprecations/constructor-deprecation-data.ts]: this migration is not part of this PR

File: projects/schematics/src/migrations/2_0/constructor-deprecations/constructor-deprecation-data.ts [hunk 2]
```diff
@@ -33,17 +38,17 @@ import { PAYMENT_METHOD_COMPONENT_MIGRATIONS } from './data/payment-method.compo
 import { PRODUCT_CAROUSEL_COMPONENT_MIGRATION } from './data/product-carousel-component.migration';
 import { PRODUCT_SERVICE_MIGRATION } from './data/product-service.migration';
 import { PROMOTION_SERVICE_MIGRATION } from './data/promotion.service.migration';
+import { REGISTER_COMPONENT_MIGRATIONS } from './data/register.component.migration';
```
Comment (tobi-or-not-tobi) [inline projects/schematics/src/migrations/2_0/constructor-deprecations/constructor-deprecation-data.ts]: this migration is not part of this PR

**Review by tobi-or-not-tobi — DISMISSED** [review]

File: projects/schematics/src/migrations/2_0/constructor-deprecations/constructor-deprecation-data.ts (hunk 1 above)
Comment (bgambocjaviniar) [reply projects/schematics/src/migrations/2_0/constructor-deprecations/constructor-deprecation-data.ts]: yup, but I believe it was just a change of position from prettier since I don't use those migration in my PR (it's from develop when I merged it to this PR)

File: projects/schematics/src/migrations/2_0/constructor-deprecations/constructor-deprecation-data.ts (hunk 2 above)
Comment (bgambocjaviniar) [reply projects/schematics/src/migrations/2_0/constructor-deprecations/constructor-deprecation-data.ts]: yup, but I believe it was just a change of position from prettier since I don't use those migration in my PR (it's from develop when I merged it to this PR)

File: projects/schematics/src/migrations/2_0/constructor-deprecations/constructor-deprecation-data.ts (hunk 1 above)
Comment (bgambocjaviniar) [reply projects/schematics/src/migrations/2_0/constructor-deprecations/constructor-deprecation-data.ts]: got moved from line 46

File: projects/schematics/src/migrations/2_0/constructor-deprecations/constructor-deprecation-data.ts (hunk 2 above)
Comment (bgambocjaviniar) [reply projects/schematics/src/migrations/2_0/constructor-deprecations/constructor-deprecation-data.ts]: got moved from line 41

**Review by psmul — DISMISSED** [review]
LGTM

**Review by LTiger14 — APPROVED** [review]

**Review by znikola — APPROVED** [review]
